src/main/java/com/openlattice/chronicle/services/CommonTasksManager.java

```diff
@@ -0,0 +1,106 @@
+package com.openlattice.chronicle.services;
+
+import com.google.common.collect.ImmutableMap;
+import com.openlattice.chronicle.constants.*;
+import com.openlattice.client.ApiClient;
+import com.openlattice.entitysets.EntitySetsApi;
+import org.apache.olingo.commons.api.edm.FullQualifiedName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+
+import static com.openlattice.chronicle.constants.AppComponent.CHRONICLE;
+import static com.openlattice.chronicle.util.ChronicleServerUtil.getParticipantEntitySetName;
+
+/**
+ * @author alfoncenzioka &lt;alfonce@openlattice.com&gt;
+ */
+public class CommonTasksManager {
+    protected static final Logger logger = LoggerFactory.getLogger( CommonTasksManager.class );
+
+    private final ApiCacheManager apiCacheManager;
+    private final EdmCacheManager edmCacheManager;
+    private final ScheduledTasksManager scheduledTasksManager;
+
+    public CommonTasksManager(
+            ApiCacheManager apiCacheManager,
+            EdmCacheManager edmCacheManager,
+            ScheduledTasksManager scheduledTasksManager ) {
+        this.apiCacheManager = apiCacheManager;
+        this.edmCacheManager = edmCacheManager;
+        this.scheduledTasksManager = scheduledTasksManager;
+    }
+
+    public UUID getPropertyTypeId( FullQualifiedName fqn ) {
+
+        return edmCacheManager.propertyTypeIdsByFQN.get( fqn );
+    }
+
+    public UUID getEntitySetId(
+            UUID organizationId,
+            AppComponent appComponent,
+            CollectionTemplateTypeName templateName,
+            String entitySetName
+    ) {
+
+        if ( organizationId == null )
+            return edmCacheManager.entitySetIdMap.getOrDefault( entitySetName, null );
+
+        Map<CollectionTemplateTypeName, UUID> templateEntitySetIdMap = scheduledTasksManager.entitySetIdsByOrgId
+                .getOrDefault( appComponent, ImmutableMap.of() )
+                .getOrDefault( organizationId, ImmutableMap.of() );
+
+        if ( templateEntitySetIdMap.isEmpty() ) {
+            logger.error( "organization {} does not have app {} installed", organizationId, appComponent );
+        }
+
+        if ( !templateEntitySetIdMap.containsKey( templateName ) ) {
+            logger.error( "app {} does not have a template {} in its entityTypeCollection",
+                    appComponent,
+                    templateName );
+        }
+
+        return templateEntitySetIdMap.get( templateName );
+    }
+
+    public UUID getParticipantEntitySetId( UUID organizationId, UUID studyId ) {
+        if ( organizationId == null ) {
+            try {
+                ApiClient apiClient = apiCacheManager.prodApiClientCache.get( ApiClient.class );
+                EntitySetsApi entitySetsApi = apiClient.getEntitySetsApi();
+
+                return entitySetsApi.getEntitySetId( getParticipantEntitySetName( studyId ) );
```

Comment:
Isn't this already stored in `entitySetIdsByOrgId`?

Reply:
No, the `entitySetIdsByOrgId` is specific to apps v2 context.